src/core/emisstate_mod.F90

```diff
@@ -74,21 +94,302 @@ module EmisState_Mod
+               if (RC /= CC_SUCCESS) then
+                  ErrMsg = 'Error allocating "EmisState%Cats%Species%EmisMapIndex"!'
+                  call Handle_Error(ErrMsg, RC, ThisLoc)
+                  return
+               endif
+
+            end do
+         end do
+      end if
+
+   end subroutine Emis_Allocate
+
+   !> \brief Allocate the total emitted species
+   !!
+   !! \details Allocate the total emitted species
+   !!
+   !! \ingroup core_modules
+   !!
+   !! \param GridState Grid State
+   !! \param EmisState Emission State
+   !! \param RC Return code
+   !!
+   !!!>
+   subroutine TotEmisSpecies_Allocate(GridState, EmisState, RC)
+
+      USE GridState_Mod, ONLY : GridStateType
+
+      IMPLICIT NONE
+
+      TYPE(GridStateType), INTENT(IN) :: GridState
+
+      TYPE(EmisStateType), INTENT(INOUT) :: EmisState
+
+      INTEGER, INTENT(OUT) :: RC
+
+      integer :: s ! Loop counter for emitted species
+      integer :: c ! Loop counter for emission Cats
+      integer :: t ! loop counter for total emitted species
+
+      integer :: nT ! total number of emitted species counter
+      integer :: nS ! number of the same emitted species in different categories
+
+      logical :: IsIn
+
+      character(len=255) :: ErrMsg
+      character(len=255) :: ThisLoc
+
+      character(len=10) ::currName
+
+      ! Initialize return code
+      RC = CC_SUCCESS
+
+      ! Initialize local variables
+      ErrMsg = ''
+      ThisLoc = ' -> at TotEmisSpecies_Allocate (in core/emisstate_mod.F90)'
+
+      ALLOCATE(EmisState%TotEmisNames(0), STAT=RC)
+      if (EmisState%nCats > 0) then
+         EmisState%nEmisTotal = 0
+         do c = 1, EmisState%nCats
+            do s = 1, EmisState%Cats(c)%nSpecies
+               currName = EmisState%Cats(c)%Species(s)%name
+               if ( ANY( EmisState%TotEmisNames == TRIM(currName) ) .eqv. .False. ) THEN
+                  EmisState%nEmisTotal = EmisState%nEmisTotal + 1
+                  EmisState%TotEmisNames = [EmisState%TotEmisNames, TRIM(currName)]
+               endif
+            end do
+         end do
+
+         ALLOCATE(EmisState%TotSpecies(EmisState%nEmisTotal), STAT=RC)
+         IF (RC /= CC_SUCCESS) THEN
+            ErrMsg = 'Error allocating "EmisState%TotSpecies"!'
+            call Handle_Error(ErrMsg, RC, ThisLoc)
+            return
+         ENDIF
+
+         do t = 1, EmisState%nEmisTotal
+            ! Fill Total Species fluxes and properties
+            EmisState%TotSpecies(t)%name = EmisState%TotEmisNames(s)
+            EmisState%TotSpecies(t)%Flux = 0. ! set all fluxes to zero
+            currName = EmisState%TotEmisNames(s)
+            do c = 1, EmisState%nCats
+               do s = 1, EmisState%Cats(c)%nSpecies
+                  if (currName == EmisState%Cats(c)%Species(s)%name) then
+                     EmisState%TotSpecies(t)%Flux(:) = EmisState%TotSpecies(t)%Flux(:) +  &
+                        EmisState%Cats(c)%Species(s)%Flux(:)
```

Comment:
I guess `EmisState%Cats(c)%Species(s)%Flux(:)` is nonzero? or at least could be?

Reply:
it should always be zero or positive definite for the emission itself no?